src/app/shared-services/trajectory.service.ts

```diff
@@ -70,7 +72,22 @@ export class TrajectoryService {
         )
 
       default:
-        return from(this.db.getFullTrajectory(id))
+        return new Observable<Trajectory>((subscriber) => {
+          // getting the trajectory from db and store it in cache
+          this.db.getFullTrajectory(id).then((trajectory) => {
+            this.trajectoryCache = trajectory
+
+            // publish trajectory
+            subscriber.next(this.trajectoryCache)
+
+            // subscribe to addPoint events
+            this.db.addPointSub.subscribe(async (point) => {
```

Comment:
We should also unsubscribe this, when the outer Observable got unsubscribed.
This can be done by calling `subscriber.add(inner)`, where `inner` is the result of this subscribe call

Reply:
Adding the inner subscription to outer in 9985a58. Thanks for the hint!